### PR TITLE
Vendor all dependencies

### PR DESCRIPTION
**Pull request by t0xicCode**
Rather than relying on the GOPATH for all dependencies, we should bundle the supported versions of those dependencies with taskcluster-cli.  And provide a nice way to update and test those dependencies.

### CODE REVIEW

**Comment by petemoore** [comment]
That's indeed a fair point. :-)

**Comment by t0xicCode** [comment]
I was actually about to add a commit that updated the readme to the PR. :)

I'll do so in a second.

File: README.md [hunk 1]
```diff
@@ -36,6 +36,22 @@ To actually build the application, simply run `make` in
 `$GOPATH/github.com/taskcluster/taskcluster-cli` which will generate the
 executable `taskcluster` in the root of the source.
 
+Dependency vendoring
+--------------------
+
+The dependencies are managed through the
+[govendor](https://github.com/kardianos/govendor) tool, but it's use should be
```
Comment (djmitche) [inline README.md]: `its`

File: README.md [hunk 2]
```diff
@@ -36,6 +36,22 @@ To actually build the application, simply run `make` in
...
+transparent when building the project. After cloning the project, running
+`govendor sync` while download the various dependencies and ensure that they
```
Comment (djmitche) [inline README.md]: `will download`

Comment (t0xicCode) [inline README.md]: Not sure where my head was...

**Comment by t0xicCode** [comment]
@djmitche ready to merge?

**Comment by djmitche** [comment]
I'm deferring to @petemoore for the technical correctness, but he said yes, so .. YES!

Please bring this up tomorrow at the meeting, too.

**Pull request merged**